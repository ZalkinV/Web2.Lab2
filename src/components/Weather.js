import React from "react";
import { getIconURL } from "../middlewares";


export default function Weather(props) {
  return (
    <div>
      <div>{props.forecast.cityName}</div>
      <img src={getIconURL(props.forecast.icon)} alt="Weather icon" />
    </div>
  );
}


