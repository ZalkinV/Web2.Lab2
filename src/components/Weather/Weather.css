.weather {
  display: grid;
  grid-template-rows: 2fr repeat(5, 1fr);
  background-color: yellow;
}

.weather > .header {
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.header > .city-name {
  font-size: 2em;
  font-display: bold;
}
