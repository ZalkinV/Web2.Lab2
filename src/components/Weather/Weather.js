import React from "react";

import { getIconURL } from "../../middlewares";
import WeatherParam from "../WeatherParam/WeatherParam";

import "./Weather.css";


export default class Weather extends React.Component {
  componentDidMount() {
    this.state.setState({isLoading: true});
    this.props.fetchForecast();
  }

  render() {
    const {
      forecast: {
        cityName,
        temperature,
        icon,
        windSpeed,
        description,
        pressure,
        humidity,
        coords: {
          lat: latitude,
          lon: longitude
        } = {}
      },
      onDelete
    } = this.props;

    return (
      <div class="weather">
        <div class="header">
          <div class="city-name">{cityName}</div>
          <div class="temperature">{temperature} &#8451;</div>
          <img src={getIconURL(icon)} alt="Weather icon" />
        </div>

        <WeatherParam name="Wind" value={`${windSpeed} m/s`} />
        <WeatherParam name="Cloudness" value={description} />
        <WeatherParam name="Pressure" value={`${pressure} hPa`} />
        <WeatherParam name="Humidity" value={`${humidity}%`} />
        <WeatherParam name="Coords" value={`${latitude}, ${longitude}`} />

        {onDelete && <button class="button" onClick={onDelete}>X</button>}
      </div>
    );}
}
