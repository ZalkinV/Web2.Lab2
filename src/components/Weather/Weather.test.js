import React from "react";
import renderer from "react-test-renderer";

import Weather from "./Weather";


describe("<Weather /> render", () => {

  const DEFAULT_FORECAST = {
    cityName: "Moscow",
    temperature: "-4",
    icon: "icon_code",
    windSpeed: 10,
    description: "nice to swim",
    pressure: 234,
    humidity: 99,
    coords: {
      lat: 40,
      lon: 40
    }
  };

  test("should has forecast and delete button", () => {
    const tree = renderer
      .create(
        <Weather
          onFetch={() => {}}
          onDelete={() => {}}
          forecast={DEFAULT_FORECAST}
        />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });
  
  test("should has forecast and no delete button", () => {
    const tree = renderer
      .create(
        <Weather
          onFetch={() => {}}
          forecast={DEFAULT_FORECAST}
        />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  test("should has <Loader />", () => {
    const tree = renderer
      .create(
        <Weather
          onFetch={() => {}}
        />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });

});