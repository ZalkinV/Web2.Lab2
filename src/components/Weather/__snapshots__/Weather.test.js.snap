// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Weather /> render should has forecast and delete button 1`] = `
<div
  className="weather"
>
  <div
    className="header"
  >
    <div
      className="city-name"
    >
      Moscow
    </div>
    <div
      className="temperature"
    >
      -4
       â„ƒ
    </div>
    <img
      alt="Weather icon"
      src="https://openweathermap.org/img/wn/icon_code.png"
    />
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Wind
    </div>
    <div
      className="value"
    >
      10 m/s
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Cloudness
    </div>
    <div
      className="value"
    >
      nice to swim
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Pressure
    </div>
    <div
      className="value"
    >
      234 hPa
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Humidity
    </div>
    <div
      className="value"
    >
      99%
    </div>
  </div>
  <div
    className="param"
  >
    <div
      className="name"
    >
      Coords
    </div>
    <div
      className="value"
    >
      40, 40
    </div>
  </div>
  <button
    className="button"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
